lib/modules/dosomething/dosomething_signup/dosomething_signup.signup_data_form.inc

```diff
@@ -309,11 +316,94 @@ function dosomething_signup_user_signup_data_form($form, &$form_state, $signup)
     '#value' => $signup->run_nid,
     '#access' => FALSE,
   );
-  $form['header'] = array(
-    '#prefix' => '<div class="modal__block"><h2 class="heading">',
-    '#markup' => $config['form_header'],
-    '#suffix' => '</h2></div>',
-  );
+
+  // If custom social share signup is turned on, use the $config['form_header'] from signup data form.
+  if ($config['custom_social_share_signup']) {
+    $form['header'] = array(
+      '#prefix' => '<div class="modal__block"><h2 class="heading">',
+      '#markup' => dosomething_helpers_get_variable('node', $signup->nid, 'social_share_custom_text'),
+      '#suffix' => '</h2></div>',
+    );
+
+    $form['copy'] = array(
+      '#prefix' => '<div class="modal__block"><p>',
+      '#markup' => dosomething_helpers_get_variable('node', $signup->nid, 'social_share_custom_description'),
+      '#suffix' => '</p></div>',
+    );
+
+    $form['custom_social_share_signup_img'] = array(
+      '#prefix' => '<img src="',
+      '#markup' => dosomething_image_get_themed_image_url(dosomething_helpers_get_variable('node', $signup->nid, 'share_image_nid'), 'landscape'),
+      '#suffix' => '">',
+    );
+
+    $custom_social_copy_variables = [
+      'custom_facebook_copy' => dosomething_helpers_get_variable('node', $signup->nid, 'custom_facebook_copy'),
+      'custom_twitter_copy' => dosomething_helpers_get_variable('node', $signup->nid, 'custom_twitter_copy'),
+      'custom_tumblr_copy' => dosomething_helpers_get_variable('node', $signup->nid, 'custom_tumblr_copy'),
+    ];
+
+    $campaign = node_load($signup->nid);
+
+    foreach ($custom_social_copy_variables as $key => $value) {
+      if ($value) {
+        $$key = $value;
+      }
+      elseif (isset($campaign->field_fact_problem['en'][0]['entity']->title)) {
+        $$key = $campaign->field_fact_problem['en'][0]['entity']->title;
```

Comment:
![image](https://cloud.githubusercontent.com/assets/105849/22706729/5a194628-ed3e-11e6-8cb7-5c094af5f60c.png)

hmmm, this is how it returns from the API which does a bit of transforming but should still be easy to extract. It should already account for the language. If it works with the other format that's cool too. Just thought it would be more straightforward since the Campaign class was easier to use in other scenarios but might not apply to this one! No worries :)